features/step_definitions/settings_steps.rb

```diff
@@ -20,5 +20,18 @@
 end
 
 Then "{provider} should have strong passwords {enabled}" do |provider, enabled|
-  assert provider.settings.strong_passwords_enabled == enabled
+  assert provider.settings.reload.strong_passwords_enabled == enabled
+end
+
+Given "{provider} should have account approval checkbox {enabled}" do |provider, enabled|
```

Comment:
Maybe name step something like 
```
Given "{provider} has only one account plan" do |provider, one|
```

I think it will be clearer both here and in scenario to read.

Reply:
Hm... but this wording would not be correct :shrug: 

I know it's confusing, but it's just simulating the conditions under which the "account approval" checkbox must be either enabled or disabled. This is defined by the following lines:

https://github.com/3scale/porta/blob/74e675416c69b5dba9847ab4d6f8da70018a3e7b/app/views/sites/usage_rules/edit.html.slim#L29-L34

https://github.com/3scale/porta/blob/74e675416c69b5dba9847ab4d6f8da70018a3e7b/app/models/settings.rb#L30-L36

So, the following outcomes are possible:

1. 1 account plan: checkbox is visible and enabled (whether `account_plans_ui_visible`* is true or false), the value is the `approval_required` value of this single plan

2. \>1 account plans: 
  a) if `account_plans_ui_visible == true`: checkbox is visible and disabled (because you are supposed to update the checkbox on the account plan explicitly)
  b) if `account_plans_ui_visible == false`: checkbox is not even visible - this scenario is not currently handled by the tests.


* `account_plans_ui_visible` is set by "Enable Account Plans" checkbox



While writing this, I realized that the code can be improved a bit, and an unnecessary step can be removed. I'll update now.

Reply:
So, I refactored this @akostadinov I think now it makes much more sense :grimacing: 

https://github.com/3scale/porta/pull/3666/commits/6054c537121703481693380271dbaed85bbbfa3f
(and then I also improved it in https://github.com/3scale/porta/pull/3666/commits/5c5f82ab91b864ac6c2d566f18853002eaabe557)

I don't like that there are 3 scenarios to check a single stupid checkbox... But probably mixing them all in one is not good either... (@josemigallas would for sure object)